testing/test_terminal.py

```diff
@@ -130,6 +130,52 @@ def test_writeline(self, pytester: Pytester, linecomp) -> None:
         assert lines[1].endswith(modcol.name + " .")
         assert lines[2] == "hello world"
 
+    def test_writeline_after_report_issue12777(
+        self, pytester: Pytester, linecomp
+    ) -> None:
+        r"""
+        This test is checking issue 12777 - where writeline is writing a new line at the first usage, due to
```

Comment:
Thanks @Oreldm for taking the time to write this, but I feel this is testing too much of the internals, which makes this test brittle and hard to maintain.

Can you try to reproduce the problem just via a `pytester.runpytest` call and checking the output? It is of course important that this test without your change fails, and passes with it.

Reply:
Thanks @nicoddemus . I agree, it felt wrong for me as well. I refactored the test.